modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -1170,6 +1185,142 @@ func (suite *KeeperTestSuite) TestWriteUpgradeOpenChannel() {
+			malleate: func() {},
+			expPanic: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			path.EndpointA.ChannelConfig.Order = types.ORDERED
+			path.EndpointB.ChannelConfig.Order = types.ORDERED
+			suite.coordinator.Setup(path)
+
+			// Need to create a packet commitment on A so as to keep it from going to OPEN if no inflight packets exist.
+			sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+			suite.Require().NoError(err)
+			packet := types.NewPacket(ibctesting.MockPacketData, sequence, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+			err = path.EndpointB.RecvPacket(packet)
+			suite.Require().NoError(err)
+
+			// send second packet from B to A
+			sequence0, err := path.EndpointB.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+			suite.Require().NoError(err)
+			packet0 := types.NewPacket(ibctesting.MockPacketData, sequence0, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+			err = path.EndpointA.RecvPacket(packet0)
+			suite.Require().NoError(err)
+
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
+			path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Ordering = types.UNORDERED
+			path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Ordering = types.UNORDERED
+
+			suite.Require().NoError(path.EndpointA.ChanUpgradeInit())
+			suite.Require().NoError(path.EndpointB.ChanUpgradeTry())
+			suite.Require().NoError(path.EndpointA.ChanUpgradeAck())
+			suite.Require().NoError(path.EndpointB.ChanUpgradeConfirm())
+
+			// Ack packets to delete packet commitments before calling WriteUpgradeOpenChannel
+			err = path.EndpointA.AcknowledgePacket(packet, ibctesting.MockAcknowledgement)
+			suite.Require().NoError(err)
+
+			err = path.EndpointB.AcknowledgePacket(packet0, ibctesting.MockAcknowledgement)
+			suite.Require().NoError(err)
+
+			ctx := suite.chainA.GetContext()
+
+			// assert that NextSeqAck is incremented to 2 because channel is still ordered
+			seq, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetNextSequenceAck(ctx, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+			suite.Require().True(found)
+			suite.Require().Equal(uint64(2), seq)
+
+			// assert that sequence (set in ChanUpgradeTry) is incremented to 2 because channel is still ordered
+			seq, found = suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetNextSequenceRecv(ctx, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
```

Comment:
Should we also be assert the counterpartyNextSequenceSend is 2 here?

Reply:
that set isnt in this PR, it's in [this one ](https://github.com/cosmos/ibc-go/pull/5460/files), the test can be updated there tho for sure!